### ISSUE TITLE
Create BatchNorm layer

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#3]
## Introduction
Create a batchNorm layer (Not spatial yet).

## Task
1. Derive new class from Base Layer
2. Create a test

## References
* [Book AI](https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/batch_norm_layer.html)
* [Cs231n Code forward](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/layers.py#L128)
* [Cs231n Code backward](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/layers.py#L264)
* [Cs231n Notebook](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/BatchNormalization.ipynb)
* [Blog 1](https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html)
* [Blog 2](http://www.computervisionblog.com/)
* [Blog 3](http://r2rt.com/implementing-batch-normalization-in-tensorflow.html)
* [Blog 4](http://ruishu.io/2016/12/27/batchnorm/)
* [Blog 5](http://torch.ch/blog/2015/07/30/cifar.html)
* [Blog 6](http://cthorey.github.io./backpropagation/)

**Comment by leonardoaraujosantos** [comment]
## Results

### Without BatchNorm
![nodropout](https://cloud.githubusercontent.com/assets/898383/22342367/11d38bae-e3ec-11e6-88b7-c9e3e2b630ca.png)

### With BatchNorm
![withbatchnorm](https://cloud.githubusercontent.com/assets/898383/22342384/1b9cc9d4-e3ec-11e6-8260-e0467aab188a.png)
